staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -343,6 +343,9 @@ type ListOptions struct {
 	TypeMeta `json:",inline"`
 
 	// A selector to restrict the list of returned objects by their labels.
+	// If you have a LabelSelector object and need to use it for this field, use
+	// FormatLabelSelector(selector) rather than selector.String() to create a
+	// string in the correct format.
```

Comment:
This documentation is intended to define the API contract be programming language agnostic. This comment is change is client-go centric. I don't think it belongs here.

Reply:
That's a fair point. Is there a place where this comment should go? Changing `String()` to return  the right thing would be a breaking change I guess, but this is a common pitfall we _should_ warn users about.

Reply:
+1 to warning users of pitfalls.

Kubernetes generates `String()` for all the generated struct bindings. The `String()`functions output go idiomatic representation of the struct. It might be an improvement to have a godoc string on all the generated `String()` functions so it's more clear what they do to help guide developers on when it is appropriate to use? I don't know how easy it would be to add that..

Letting developers know of the presence of the `FormatLabelSelector` function would be great. But again, I'm not sure where to put it.

Sorry I can't be more help. Suggestion welcome.